generator/internal/codegen_utils.h

```diff
@@ -63,6 +65,25 @@ std::string ServiceNameToFilePath(absl::string_view service_name);
  */
 std::string ProtoNameToCppName(absl::string_view proto_name);
 
+enum class NamespaceType { kNormal, kInternal };
+
+/**
+ * Builds namespace hierarchy.
+ */
+std::vector<std::string> BuildNamespaces(
+    std::map<std::string, std::string> const& vars,
+    NamespaceType ns_type = NamespaceType::kNormal);
+
+/**
+ * Validates command line arguments passed to the microgenerator.
+ *
+ * Command line arguments can be passed from the protoc command line via:
+ * '--cpp_codegen_opt=key=value'. This can be specified multiple times to
```

Comment:
So the `parameters` argument should be a string that looks like `"--cpp_codegen_opt=key=value"` or `"--cpp_codegen_opt=key=value --cpp_codegen_opt=key=value --cpp_codegen_opt=key=value"` ?